packages/lodestar/src/db/api/beacon/repositories/block.ts

```diff
@@ -1,72 +1,46 @@
 import {Slot, SignedBeaconBlock} from "@chainsafe/lodestar-types";
 import {IBeaconConfig} from "@chainsafe/lodestar-config";
 
-import {BulkRepository} from "../repository";
-import {ChainRepository} from "./chain";
 import {IDatabaseController} from "../../../controller";
-import {Bucket, encodeKey} from "../../../schema";
+import {Bucket, encodeKey} from "../../schema";
+import {Repository} from "./abstract";
+import {ChainRepository} from "./chain";
 
-export class BlockRepository extends BulkRepository<SignedBeaconBlock> {
+/**
+ * Blocks by root
+ *
+ * Used to store unfinalized blocks
+ */
+export class BlockRepository extends Repository<Uint8Array, SignedBeaconBlock> {
 
   private chain: ChainRepository;
 
   public constructor(
     config: IBeaconConfig,
-    db: IDatabaseController,
+    db: IDatabaseController<Buffer, Buffer>,
     chain: ChainRepository) {
     super(config, db, Bucket.block, config.types.SignedBeaconBlock);
     this.chain = chain;
   }
 
+  /**
+   * Id is hashTreeRoot of unsigned BeaconBlock
+   */
   public getId(value: SignedBeaconBlock): Uint8Array {
     return this.config.types.BeaconBlock.hashTreeRoot(value.message);
   }
 
-  public async set(id: Uint8Array, value: SignedBeaconBlock): Promise<void> {
+  public async put(id: Uint8Array, value: SignedBeaconBlock): Promise<void> {
     await Promise.all([
-      this.db.put(encodeKey(Bucket.blockSlotRefs, value.message.slot), id),
-      this.db.put(encodeKey(Bucket.blockRootRefs, id), this.config.types.Slot.serialize(value.message.slot)),
-      super.set(id, value)
+      this.db.put(encodeKey(Bucket.blockSlotRefs, value.message.slot), id as Buffer),
```

Comment:
Actually, in non-finalized blocks, there can be more than one block for one slot. I think we should add root to array of roots belonging to that slot. Maybe even have separate repository to handle those.

Reply:
Yeah, I would have removed it entirely but its being used right now. Trying to limit the damage of a single PR XD